src/main/java/seedu/address/model/event/EventDate.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.model.event;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents an Event's date in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDate(String)}
+ */
+public class EventDate {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Date should be in the format of YYYY-MM-DD, and it should not be blank";
+
+    public static final DateTimeFormatter LOCAL_DATE_FORMATTER = DateTimeFormatter.ofPattern("YYYY-MM-DD");
+    public final LocalDate date;
+
+    /**
+     * Constructs a {@code Name}.
+     *
+     * @param dateString A valid date in String.
+     */
+    public EventDate(String dateString) {
+        requireNonNull(dateString);
+        checkArgument(isValidDate(dateString), MESSAGE_CONSTRAINTS);
+        date = LocalDate.parse(dateString);
+    }
+
+    /**
+     * Returns true if a given date is a valid date.
+     */
+    public static boolean isValidDate(String test) {
+        try {
+            LocalDate.parse(test);
+            return true;
```

Comment:
I think here some a constraint can be enforced, like checking for 30th february is an invalid date. 
An approach i read up could be 
`DateTimeFormatter formatter = DateTimeFormatter.ofPattern( "YYYY-MM-DD").withResolverStyle(ResolverStyle.STRICT); `
`LocalDate.parse(test, formatter);`
Reference source: https://stackoverflow.com/questions/74249440/validate-a-date-in-a-string-java-block-errors-like-feb-30

Reply:
Oh I was doing it under the impression that LocalDate.parse wouldn't be able to parse that already?
<img width="1378" alt="image" src="https://github.com/AY2324S1-CS2103T-F11-1/tp/assets/56021409/6db8c6af-0f4b-4fd3-8187-d59a5a4a20e5">

Reply:
Oooo okay nvm then :0